api/src/main/java/org/apache/gravitino/authorization/Privilege.java

```diff
@@ -39,6 +40,14 @@ public interface Privilege {
    */
   Condition condition();
 
+  /**
+   * If a metadata object supports a privilege, the privilege will return true. Otherwise, false.
+   *
+   * @param type The metadata object type.
+   * @return It will return true if a metadata object supports a privilege. Otherwise, false.
+   */
+  boolean supportsMetadataObject(MetadataObject.Type type);
```

Comment:
If the method name is `supportsMetadataObject`, you'd better rename it to `supportsMetadataObjectWithType`

Reply:
or change the parameter type to `MetadataObjec`

Reply:
change to `supportsMetadataObjectType`.